src/wallet/wallet.cpp

```diff
@@ -1684,7 +1697,15 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, CBlock
             if (pindex == pindexStop) {
                 break;
             }
-            pindex = chainActive.Next(pindex);
+            {
+                LOCK(cs_main);
+                pindex = chainActive.Next(pindex);
+                if (pindex && !chainActive.Contains(pindex)) break;
```

Comment:
This line is a noop - chainActive.Next() is never going to return something non-nullptr that isn't on the best chain.

Reply:
Oops. This is a rebase issue, ... removing this line now.